test:
from pyGoogleTrendsCsvDownloader import pyGoogleTrendsCsvDownloader r = pyGoogleTrendsCsvDownloader(login, password') r.get_csv(cat='0-958',date='1/2006 12m', geo='US-ME-500')
error:
Traceback (most recent call last): File "test.py", line 6, in <module> r.get_csv(cat='0-958',date='1/2006 12m', geo='US-ME-500') File "/home/savril/workspace/google-trends-csv-downloader/pyGoogleTrendsCsvDownloader.py", line 134, in get_csv myFile = open('trends_%s.csv' % '_'.join(['%s-%s' % (key, value) for (key, value) in kwargs.items()]), 'w') IOError: [Errno 2] No such file or directory: 'trends_date-1/2006 12m_geo-US-ME-500_cat-0-958.csv'

--- a/pyGoogleTrendsCsvDownloader.py
+++ b/pyGoogleTrendsCsvDownloader.py
@@ -14,6 +14,17 @@ import sys
 from cookielib import Cookie, CookieJar
 from StringIO import StringIO
 
+# Copied from Django
+def slugify(value):
+    """
+    Converts to lowercase, removes non-word characters (alphanumerics and
+    underscores) and converts spaces to hyphens. Also strips leading and
+    trailing whitespace.
+    """
+    import unicodedata, re
+    value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
+    value = re.sub('[^\w\s-]', '', value).strip().lower()
+    return unicode(re.sub('[-\s]+', '-', value))
 
 class pyGoogleTrendsCsvDownloader(object):
     '''
@@ -131,6 +142,6 @@ class pyGoogleTrendsCsvDownloader(object):
         else:
             data = r.read()
         
-        myFile = open('trends_%s.csv' % '_'.join(['%s-%s' % (key, ''.join([char for char in value if char.isalnum()])) for (key, value) in kwargs.items()]), 'w')
+        myFile = open('trends_%s.csv' % '_'.join(['%s-%s' % (key, slugify(unicode(value))) for (key, value) in kwargs.items()]), 'w')
         myFile.write(data)
         myFile.close()

